fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java

```diff
@@ -218,9 +218,96 @@ public static void convertColumnFilter(ScalarOperator predicate, Map<String, Par
             return;
         }
 
+        // Fast path: build range for date_trunc(...) on expression-partitioned tables; if handled, return.
+        if (predicate instanceof BinaryPredicateOperator
+                && predicate.getChild(0) instanceof CallOperator
+                && buildDateTruncRange((BinaryPredicateOperator) predicate, result, table)) {
+            return;
+        }
+
         predicate.accept(COLUMN_FILTER_VISITOR, result);
     }
 
+    // Build a PartitionColumnFilter range for date_trunc predicates. Returns true if handled.
+    private static boolean buildDateTruncRange(BinaryPredicateOperator predicate,
+                                               Map<String, PartitionColumnFilter> result,
+                                               Table table) {
+        if (!(table instanceof OlapTable)) {
+            return false;
+        }
+        if (!(predicate.getChild(0) instanceof CallOperator call) ||
+                !(predicate.getChild(1) instanceof ConstantOperator rhsConst)) {
+            return false;
+        }
+        if (!FunctionSet.DATE_TRUNC.equals(call.getFnName())) {
+            return false;
+        }
+        PartitionInfo pinfo = ((OlapTable) table).getPartitionInfo();
+        if (!(pinfo instanceof ExpressionRangePartitionInfo)) {
```

Comment:
how about `ExpressionRangePartitionInfoV2`? ExpressionRangePartitionInfo seems depredated?

Reply:
seems ExpressionRangePartitionInfoV2 can't use this optimization. I just follow this [check](https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java#L286)